docs/changes/stats/16562.feature.rst

```diff
@@ -0,0 +1 @@
+Adds type annotations to all function (public and private, including class methods) call signatures in ``stats``.
```

Comment:
This is a bit TBD, but this should perhaps be `16562.other.rst`?  Though I can also see the case for `.feature`... See https://github.com/astropy/astropy/issues/16569#issuecomment-2169190234 - better wait for discussion before changing it!

Reply:
Thanks Marten! I'm happy for this to be `other` rather than `feature`. Since no one has objected here or in #16569, I've changed it to `other` in [9597925](https://github.com/astropy/astropy/pull/16562/commits/95979257f85db21e763acf9ad0a6c58cbe1e4fce)